firmware/user/modes/mode_flight.c

```diff
@@ -795,86 +940,178 @@ static bool ICACHE_FLASH_ATTR flightRender(void)
+        int mdlct = 0;
+
+/////////////////////////////////////////////////////////////////////////////////////////
+////GAME LOGIC GOES HERE (FOR COLLISIONS/////////////////////////////////////////////////
+
+        int i;
+        for( i = 0; i < tflight->enviromodels;i++ )
+        {
+            tdModel * m = tflight->environment[i];
+
+            int label = m->label;
+            int draw = 1;
+            if( label )
+            {
+                draw = 0;
+                if( label >= 100 && (label - 100) == tflight->ondonut )
+                {
+                    draw = 1;
+                    if( tdDist( tflight->planeloc, m->center ) < 130 )
+                    {
+                        flightLEDAnimate( FLIGHT_LED_DONUT );
+                        tflight->ondonut++;
+                    }
+                }
+                //bean? 1000... groupings of 8.
+                int beansec = ((label-1000)/10);
+
+                if( label >= 1000 && ( beansec == tflight->ondonut || beansec == (tflight->ondonut-1) || beansec == (tflight->ondonut+1)) )
+                {
+                    if( ! (tflight->beangotmask[beansec] & (1<<((label-1000)%10))) )
+                    {
+                        draw = 1;
+
+                        if( tdDist( tflight->planeloc, m->center ) < 100 )
+                        {
+                            tflight->beans++;
+                            tflight->beangotmask[beansec] |= (1<<((label-1000)%10));
+                            flightLEDAnimate( FLIGHT_LED_BEAN );
+                        }
+
+                    }
+                }
+                if( label == 999 ) //gazebo
+                {
+                    draw = 1;
+                    if( flight->mode != FLIGHT_GAME_OVER && tdDist( tflight->planeloc, m->center ) < 200 && tflight->ondonut == 14)
+                    {
+                        flightLEDAnimate( FLIGHT_LED_ENDING );
+                        tflight->frames = 0;
+                        tflight->wintime = tflight->timer;
+                        tflight->mode = FLIGHT_GAME_OVER;
+                    }
+                }
+            }
+
+            if( draw == 0 ) continue;
+
+            int r = tdModelVisibilitycheck( m );
+            if( r < 0 ) continue;
+            mrp[mdlct].model = m;
+            mrp[mdlct].mrange = r;
+            mdlct++;
+        }
+
+        //Painter's algorithm
+        qsort( mrp, mdlct, sizeof( struct ModelRangePair ), mdlctcmp );
+
+        for( i = 0; i < mdlct; i++ )
+        {
+            tdModel * m = mrp[i].model;
+            int label = m->label;
+            int draw = 1;
+            if( label )
+            {
+                draw = 0;
+                if( label >= 100 && label < 999 )
+                {
+                    draw = 2; //All donuts flash on.
+                }
+                if( label >= 1000 )
+                {
+                    draw = 3; //All beans flash-invert
```

Comment:
Found it. These should be inverted on a more steady clock, rather than on each render.